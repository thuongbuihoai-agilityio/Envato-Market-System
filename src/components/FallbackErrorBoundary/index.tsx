import { Text, VStack } from '@chakra-ui/react';
import { memo } from 'react';
import { FallbackProps } from 'react-error-boundary';
import areEqual from 'react-fast-compare';

const ErrorComponent = ({ error }: FallbackProps): JSX.Element => (
  <VStack justifyContent="center" minH="calc(100vh - 200px)">
    <Text fontSize="lg" fontWeight="bold" color="text.primary">
      An error has been occurred!!
    </Text>
    <Text fontSize="3xl" fontWeight="bold" color="danger.400">
      {error?.message || (error as string)}
    </Text>
  </VStack>
);

const FallbackErrorBoundary = memo(ErrorComponent, areEqual);

export default FallbackErrorBoundary;
