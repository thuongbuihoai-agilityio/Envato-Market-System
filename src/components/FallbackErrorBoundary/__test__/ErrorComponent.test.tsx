import { render, screen } from '@testing-library/react';
import FallbackErrorBoundary from '..';

const mockErrorMessage = 'mock error';

const setup = () =>
  render(
    <FallbackErrorBoundary
      error={mockErrorMessage}
      resetErrorBoundary={jest.fn()}
    />,
  );

describe('FallbackErrorBoundary test cases', () => {
  it('should render correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('should have the default error title', async () => {
    setup();

    const defaultErrorTitle =
      screen.getByTestId<HTMLParagraphElement>('error-title');

    expect(defaultErrorTitle.textContent).toEqual(
      'An error has been occurred!!',
    );
  });

  it('should match the input error message', async () => {
    setup();

    const inputErrorMessage =
      screen.getByTestId<HTMLParagraphElement>('error-message');

    expect(inputErrorMessage.textContent).toEqual(mockErrorMessage);
  });
});
