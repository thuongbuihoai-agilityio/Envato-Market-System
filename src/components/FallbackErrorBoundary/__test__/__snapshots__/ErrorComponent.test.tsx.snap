// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FallbackErrorBoundary test cases should render correctly 1`] = `
<div>
  <div
    class="chakra-stack css-1pkffy6"
  >
    <p
      class="chakra-text css-oz7p8m"
      data-testid="error-title"
    >
      An error has been occurred!!
    </p>
    <p
      class="chakra-text css-1ly9ogx"
      data-testid="error-message"
    >
      mock error
    </p>
  </div>
</div>
`;
